Various go vet and go fix fixes

gfr10598 opened the pull request:
This PR cleans up a number of go vet warnings.  It also fixes all the "context" imports that used an obsolete version of context.

I suspected a possible ipv6 triple colon bug in the geo_annotation.go code related to cip and sip (client and server ip addresses).  Added a unit test to try to expose it, but it seems that it is WAI.  Left the test there anyway.

After these changes, etl passes the following vet checks.
go tool vet --shadow .
go vet ./annotation/... ./appengine/... ./fake/... ./metrics/... ./schema/... ./task/... ./web100/...
go vet -assign -atomic -bool -buildtags -cgocall -copylocks -httpresponse -methods -nilfunc -printf -rangeloops -shift -structtags -tests -unreachable -unsafeptr -unusedresult ./bq/... ./etl/... ./parser/... ./storage/...

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/m-lab/etl/497)
<!-- Reviewable:end -->

yachang commented:
Review status: 0 of 16 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

*[cmd/task_client/task_client.go, line 9 at r1](https://reviewable.io/reviews/m-lab/etl/497#-LAxeJzo1eb3FK7rMYSk:-LAxeJzo1eb3FK7rMYSl:bvxrwem) ([raw file](https://github.com/m-lab/etl/blob/d2f037213b6a86f636637e97ad33d31f9775d6cf/cmd/task_client/task_client.go#L9)):*
> ```Go
> 	"time"
> 
> 	"context"
> ```

context move to above?

---


*Comments from [Reviewable](https://reviewable.io/reviews/m-lab/etl/497)*
<!-- Sent from Reviewable.io -->

yachang commented:
Review status: 0 of 16 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[annotation/geo.go, line 110 at r2](https://reviewable.io/reviews/m-lab/etl/497#-LAxidJ_6wpuOEbVdWCv:-LAxidJ_6wpuOEbVdWCw:b3dgn7d) ([raw file](https://github.com/m-lab/etl/blob/11d714f6f31a7b9958ae1c3b4d40e9a5d707294b/annotation/geo.go#L110)):*
> ```Go
> 			continue
> 		}
> 		ip, _ = web100.NormalizeIPv6(ip)
> ```

I would prefer to use a different var name (such as "normalizedIP" ) instead of reuse "ip" here

---


*Comments from [Reviewable](https://reviewable.io/reviews/m-lab/etl/497)*
<!-- Sent from Reviewable.io -->

yachang commented:
Review status: 0 of 16 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[bq/insert.go, line 28 at r2](https://reviewable.io/reviews/m-lab/etl/497#-LAxm-PdDh8d10lRTxrH:-LAxm-PdDh8d10lRTxrI:bfyktvn) ([raw file](https://github.com/m-lab/etl/blob/11d714f6f31a7b9958ae1c3b4d40e9a5d707294b/bq/insert.go#L28)):*
> ```Go
> 
> 	"cloud.google.com/go/bigquery"
> 	"context"
> ```

move it to above

---


*Comments from [Reviewable](https://reviewable.io/reviews/m-lab/etl/497)*
<!-- Sent from Reviewable.io -->

gfr10598 commented:
Also added go-pre-commit, which contains the passing golint and go vet tests.

---

Review status: 0 of 16 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[annotation/geo.go, line 110 at r2](https://reviewable.io/reviews/m-lab/etl/497#-LAxidJ_6wpuOEbVdWCv:-LAxrs365W6Vmi2i-Za1:butg498) ([raw file](https://github.com/m-lab/etl/blob/11d714f6f31a7b9958ae1c3b4d40e9a5d707294b/annotation/geo.go#L110)):*
<details><summary><i>Previously, yachang wrote…</i></summary><blockquote>

I would prefer to use a different var name (such as "normalizedIP" ) instead of reuse "ip" here
</blockquote></details>

Done.  Actually added a new slice, so that we don't have to call NormalizeIPv6 twice, and also changed to avoid the range copy. 

---

*[bq/insert.go, line 28 at r2](https://reviewable.io/reviews/m-lab/etl/497#-LAxm-PdDh8d10lRTxrH:-LAxswy-DAeLr_hxE4x1:b-896fix) ([raw file](https://github.com/m-lab/etl/blob/11d714f6f31a7b9958ae1c3b4d40e9a5d707294b/bq/insert.go#L28)):*
<details><summary><i>Previously, yachang wrote…</i></summary><blockquote>

move it to above
</blockquote></details>

Done.

---

*[cmd/task_client/task_client.go, line 9 at r1](https://reviewable.io/reviews/m-lab/etl/497#-LAxeJzo1eb3FK7rMYSk:-LAxslK51e5z5f9lIl0A:b-896fix) ([raw file](https://github.com/m-lab/etl/blob/d2f037213b6a86f636637e97ad33d31f9775d6cf/cmd/task_client/task_client.go#L9)):*
<details><summary><i>Previously, yachang wrote…</i></summary><blockquote>

context move to above?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/m-lab/etl/497#-:-LAxt4u5-70XVrhc12qO:b-a0uoug)*
<!-- Sent from Reviewable.io -->

yachang commented:
Review status: 0 of 17 files reviewed at latest revision, 3 unresolved discussions.

---

*[storage/storage.go, line 24 at r2](https://reviewable.io/reviews/m-lab/etl/497#-LAxv9S_55Vo4FKqgTeS:-LAxv9SaCZWdYtODx1fc:b-7ropln) ([raw file](https://github.com/m-lab/etl/blob/11d714f6f31a7b9958ae1c3b4d40e9a5d707294b/storage/storage.go#L24)):*
> ```Go
> 	"github.com/m-lab/etl/metrics"
> 
> 	"context"
> ```

ditto

---


*Comments from [Reviewable](https://reviewable.io/reviews/m-lab/etl/497)*
<!-- Sent from Reviewable.io -->

yachang commented:
Review status: 0 of 17 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[annotation/geo.go, line 114 at r3](https://reviewable.io/reviews/m-lab/etl/497#-LAy3lFODg6CxLtBEX6u:-LAy3lFODg6CxLtBEX6v:b4ndp68) ([raw file](https://github.com/m-lab/etl/blob/8ad78900a07a9a5663a4fa79e8ea71740f59b72e/annotation/geo.go#L114)):*
> ```Go
> 		normalized[i], err = web100.NormalizeIPv6(ips[i])
> 		if err != nil {
> 			log.Println(err)
> ```

Anyway to track this kind of error with prometheus? 

---


*Comments from [Reviewable](https://reviewable.io/reviews/m-lab/etl/497)*
<!-- Sent from Reviewable.io -->

gfr10598 commented:
Review status: 0 of 17 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[annotation/geo.go, line 114 at r3](https://reviewable.io/reviews/m-lab/etl/497#-LAy3lFODg6CxLtBEX6u:-LAyK5CX7-7g53kd5bYG:b-hcupjf) ([raw file](https://github.com/m-lab/etl/blob/8ad78900a07a9a5663a4fa79e8ea71740f59b72e/annotation/geo.go#L114)):*
<details><summary><i>Previously, yachang wrote…</i></summary><blockquote>

Anyway to track this kind of error with prometheus? 
</blockquote></details>

Done.  Also updated these to be warnings instead of errors.

---

*[storage/storage.go, line 24 at r2](https://reviewable.io/reviews/m-lab/etl/497#-LAxv9S_55Vo4FKqgTeS:-LAyKDOg4DNGmhjqkYO_:b-896fix) ([raw file](https://github.com/m-lab/etl/blob/11d714f6f31a7b9958ae1c3b4d40e9a5d707294b/storage/storage.go#L24)):*
<details><summary><i>Previously, yachang wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/m-lab/etl/497)*
<!-- Sent from Reviewable.io -->

yachang commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 17 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io/reviews/m-lab/etl/497#-:-LAye8Lh8mpxR_xfhUC4:bnfp4nl)*
<!-- Sent from Reviewable.io -->
